transform/lookups.go

```diff
@@ -6,22 +6,68 @@ type lookup map[int]string
 
 type lookups struct {
 	motives lookup
+	cities  lookup
 }
 
-func newLookups() lookups {
-	return lookups{motives: make(map[int]string)}
+func newLookups(d string) (lookups, error) {
+	var ls []lookup
+	var r lookups
+	srcs := []sourceType{motives, cities}
+	for _, src := range srcs {
+		paths, err := PathsForSource(src, d)
+		if err != nil {
+			return r, fmt.Errorf("error finding sources for %s: %w", string(src), err)
+		}
+		for _, p := range paths {
+			z, err := newArchivedCSV(p, separator)
+			if err != nil {
+				return r, fmt.Errorf("error creating archivedCSV for %s: %w", p, err)
+			}
+			l, err := z.toLookup()
+			if err != nil {
+				return r, fmt.Errorf("error creating %s lookup table: %w", string(src), err)
+			}
+			z.close()
```

Comment:
Garantir que o z.close() é invocado, mesmo em caso de erro. Um princípio de sugestão segue abaixo (ainda faltam modificações, mas dá para ter uma ideia)

```suggestion
                        func() (lookups, error) {
			z, err := newArchivedCSV(p, separator)
			if err != nil {
				return nil, fmt.Errorf("error creating archivedCSV for %s: %w", p, err)
			}
			defer z.close()
			l, err := z.toLookup()
			if err != nil {
				return nil, fmt.Errorf("error creating %s lookup table: %w", string(src), err)
			}
			return l,nil
			}()
```

Reply:
Ah… boa! Envolver numa função anônima… eu estava em dúvida sobre deixar vários `defer` dentro do loop (que só seriam fechados ao final da função, não a cada iteração do loop), mas tua ideia resolve isso! Obrigado. Mais tarde vou editar esse trecho : )

Reply:
Acabei extraindo a lógica em outra função, achei que ficou melhor 99360427efbc1d78e0abd0e2d52e4042617b045e :sparkles: